### PR TITLE
HACS Data Refactor

### LINKED ISSUE #6: KeyError: 'name'
## Version of the custom_component

VERSION = "0.2.1"

## Configuration

```yaml
readme:
```

## Describe the bug
Key error with hacs.repositories file.

## Debug log

```text
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 130, in handle_call_service
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1260, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1295, in _execute_service
    await handler.func(service_call)
  File "/config/custom_components/readme/__init__.py", line 168, in service_generate
    hacs_components = get_hacs_components(hass)
  File "/config/custom_components/readme/__init__.py", line 208, in get_hacs_components
    "name": get_repository_name(repository),
  File "/config/custom_components/readme/__init__.py", line 221, in get_repository_name
    name = repository["repository_manifest"]["name"]
KeyError: 'name'
```

### PR DESCRIPTION
**Pull request by steve-gombos**
**Dependent on the next release of HACS.**

Refactored to use the new HACS updates instead of parsing the files from .storage.  

- Fixes #6

### CODE REVIEW

File: custom_components/readme/__init__.py [hunk 1]
```diff
@@ -185,50 +185,84 @@ async def service_generate(call):
...
+    except ImportError:
+        return []
+
+    hacs = get_hacs()
+    repositories = hacs.repositories
+    installed = []
+
+    for repo in repositories:
+        if repo.data.installed:
+            installed.append(
```
Comment (ludeeus) [inline custom_components/readme/__init__.py:201]: use `installed.append(repo.to_json())` instead, that is more dynamic

Comment (steve-gombos) [reply custom_components/readme/__init__.py:201]: Looks like this is actually `repo.data.to_json()`  Just pushed an update with this.

Comment (steve-gombos) [reply custom_components/readme/__init__.py:201]: Shows up on my history, but looks like github may be having some issues on the backend.

https://github.com/steve-gombos/readme/commit/f515461fd691047c335e7d6bb1b7bdc36458b35e

Comment (ludeeus) [reply custom_components/readme/__init__.py:201]: Right, my bad. xxx.data.to_json() is the correct way
GitHub is having issues ATM, if it does not update the PR in an hour or so, just push an empty commit ty retrigger the sync

Comment (steve-gombos) [reply custom_components/readme/__init__.py]: Looks like it showed up.

**Pull request merged**